haystack/utils/device.py

```diff
@@ -389,7 +389,7 @@ def has_multiple_devices(self) -> bool:
         return self._multiple_devices is not None
 
     @property
-    def first_device(self) -> Optional[Device]:
+    def first_device(self) -> Optional["ComponentDevice"]:
```

Comment:
Hey @shadeMe let me know if the doc string here should be updated. I was unsure if I should have updated it to something like 
```
python
Return either the single component device or the first component device in the device map, if any.
```